# Final Project Data Preparation 

In [11]:
#import mypytable 
from mypytable import MyPyTable 
import dataPrepUtils as dputils
import importlib

## For Salaries Dataset, Calculate Average Career Salary for each Player


## Load Dataset from file

In [12]:
#load data from file 
salaries = MyPyTable()
salaries.load_from_file("salaries_1985to2018.csv")

## Calculate Average Yearly Salary for each Player in the Dataset

In [13]:
# convert data to numeric
salaries.convert_to_numeric()

#get averages 
avg_header, avg_salaries = dputils.calculate_avg_salaries(salaries.data)

## Write avg Salaries to a new File 

In [14]:
#new pytable instance
avg_salaries_table = MyPyTable(avg_header, avg_salaries)

#write to new file 
avg_salaries_table.save_to_file("avg-player-salaries.txt")


## Load Player Data from File

In [15]:
players = MyPyTable()
players.load_from_file("players.csv")

## Perform Inner Join on Players and Average Salaries 

In [16]:
avg_players = players.perform_inner_join(avg_salaries_table, ["id"])


## Clean Data by Removing Instances with Missing Values 


In [17]:
avg_players.remove_rows_with_missing_values()

## Remove Players Drafted Before 2000 and Players who Played Less than 82 Career Games

In [18]:
updated_players = dputils.remove_players(avg_players.data, avg_players.column_names)
avg_players.data = updated_players

## Write Updated Player List to File

In [19]:
avg_players.save_to_file("cleaned-data.txt")

## Find Discretization Ranges for Salaries

In [20]:
importlib.reload(dputils)

#get_column for salaries
avg_players.convert_to_numeric()
salaries = avg_players.get_column("avg_salary")

#get bins for salaries
salary_cutoffs = dputils.compute_equal_width_cutoffs(salaries, 10)
print(salary_cutoffs)

[65000.0, 1605176.1, 3145352.2, 4685528.3, 6225704.4, 7765880.5, 9306056.6, 10846232.7, 12386408.8, 13926584.9, 15466761.0]


## Discretized Ranges for NBA Salaries (US Dollars per Year)
1. < 1605176.1
2. 1605176.1 - 3145352.2
3. 3145352.2 - 4685528.3
4. 4685528.3 - 6225704.4
5. 6225704.4 - 7765880.5
6. 7765880.5 - 9306056.6
7. 9306056.6 - 10846232.7
8. 10846232.7 - 12386408.8
9. 12386408.8 - 13926584.9 
10. 13926584.9 <
